app/models/team.rb

```diff
@@ -63,10 +63,21 @@ def in_current_season
       where(season: Season.current)
     end
 
+    def by_season_year(year)
+      where(season: Season.where(name: year))
+    end
+
+    def select_teams_by_season_year(year)
```

Comment:
This looks like a highly specialised method. Since the `by_season_year` returns an active relation, you could just add `.selected` after that. Let us go that route and remove this method.

Or if that doesn't work, try rephrasing the `selected` method as a `scope`.